### PR TITLE
Significantly simplify generic example

### LINKED ISSUE #26320: Misleading section on angle-brackets for specializing traits in `Generic functions` section
In the [reference.md](https://github.com/rust-lang/rust/blob/master/src/doc/reference.md) in the section [Generic functions](https://doc.rust-lang.org/nightly/reference.html#generic-functions) it is stated that

> Each type parameter must be explicitly declared, in an **angle-bracket**-enclosed, comma-separated list following the function name.

(emphasis mine)

But the example shows a type parameter on `Fn`, which requires parenthesis-enclosed type parameters rather than angle-brackets. `geofft` mentioned on IRC that this might be some special case to open the possibility to easier change the `Fn` traits at a later time (if I understood correctly), but I found it confusing to see a specialization using parenthesis without ever having heard they existed.

### PR DESCRIPTION
**Pull request by steveklabnik**
Fixes #26320

### CODE REVIEW

File: src/doc/reference.md [hunk 1]
```diff
@@ -928,15 +928,10 @@ A _generic function_ allows one or more _parameterized types_ to appear in its
...
-}
-fn map<T, U, F>(seq: &[T], f: F) -> Vec<U> where T: Copy, U: Copy, F: Fn(T) -> U {
-    let mut acc = vec![];
-    for elt in seq { acc.push(f(*elt)); }
-    acc
-}
+```rust,ignore
+// foo is generic over A and B
+
+fn foo<A, B>(x: A, y: B) {
```
Comment (jfager) [inline src/doc/reference.md]: This change removes the use of `where` syntax referred to in the following paragraph

Comment (steveklabnik) [inline src/doc/reference.md]: argh I read it and still missed that

Comment (steveklabnik) [inline src/doc/reference.md]: fixed

Comment (jfager) [inline src/doc/reference.md]: Still referring to `iter` two paragraphs down.

Comment (steveklabnik) [inline src/doc/reference.md]: sigh. i will open a new PR to fix that.

**Pull request merged**